### PR TITLE
GRIM: improve RTL support for Hebrew fan translation

### PR DESCRIPTION
**Pull request by BLooperZ**
This amends #2880

### CODE REVIEW

**Comment by orgads** [comment]
Can you squash the commits?

**Comment by BLooperZ** [comment]
squashed

File: engines/grim/textobject.cpp [hunk 1]
```diff
@@ -278,16 +279,11 @@ void TextObject::setupText() {
 			nextLinePos = message.size();
 			cutLen = nextLinePos;
 		}
-		Common::String currentLine;
+		Common::String currentLine(message.c_str(), message.c_str() + nextLinePos);
```
Comment (orgads) [inline engines/grim/textobject.cpp:282]: Needless copy for RTL (and the comment is now redundant). Suggestion:

```c++
		Common::String currentLine;
		if (g_grim->getGameLanguage() == Common::HE_ISR)
			currentLine = Common::convertBiDiString(currentLine, Common::kWindows1255);
		else
			currentLine = Common::String(message.c_str(), message.c_str() + nextLinePos);
```

**Review by orgads — CHANGES_REQUESTED** [review]

File: engines/grim/textobject.cpp (hunk 1 above)
Comment (BLooperZ) [reply engines/grim/textobject.cpp:282]: please note that coverBiDiString requires the created string (as first parameter), so the copy is needed

Comment (orgads) [reply engines/grim/textobject.cpp:282]: Ah, right. Sorry for the noise :)

**Review by orgads — APPROVED** [review]

**Comment by orgads** [comment]
Thank you. Merging.

**Pull request merged**